### PR TITLE
Add example `Dockerfile`. Improve error messages

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Add example `Dockerfile`
* Improve error messages
* Fix `helmfile` base component path

## why

* Example `Dockerfile` to start using `atmos` in a Docker container
* Better error handling
* `helmfile` base component path did not take into account base component (it tried to use the derived component name in `components/helmfile)

### CODE REVIEW

**Review by osterman — APPROVED** [review]

File: pkg/config/utils.go [hunk 1]
```diff
@@ -247,6 +248,83 @@ func checkConfig() error {
 	return nil
 }
 
+// processBasePath adjusts the current working dir by making it relative to the provided base path
+// NOTE: this function is not in use, we leave it here (in utils) in case we need it in the future
+func processBasePath() error {
```
Comment (osterman) [inline pkg/config/utils.go:253]: I wonder if the simple approach terraform took with `chdir` would be better? There's a lot of logic computing the absolute path. Is it necessary over the simple chdir?

cc: @Nuru

Comment (osterman) [reply pkg/config/utils.go:253]: https://www.terraform.io/cli/commands#switching-working-directory-with-chdir

Comment (aknysh) [reply pkg/config/utils.go:253]: the function is not in use, we can delete it

File: pkg/config/config.go [hunk 2]
```diff
@@ -280,6 +247,11 @@ func ProcessConfig(configAndStacksInfo ConfigAndStacksInfo) error {
 	} else {
 		// The stack is a logical name
 		// Check if it matches the pattern specified in 'StackNamePattern'
+		if len(Config.Stacks.NamePattern) == 0 {
+			errorMessage := "\nStack name pattern must be provided and must be not empty. Check the CLI config in 'atmos.yaml'"
```
Comment (johncblandii) [inline pkg/config/config.go:251]:
```suggestion
			errorMessage := "\nStack name pattern must be provided and must not be empty. Check the CLI config in 'atmos.yaml'"
```

**Review by johncblandii — COMMENTED** [review]
Just one thought there.

**Review by max-lobur — APPROVED** [review]

File: examples/complete/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ARG TF_1_0_VERSION=1.1.3
+RUN apt-get update && apt-get install -y -u --allow-downgrades \
+  terraform-1="${TF_1_0_VERSION}-*" && \
+  update-alternatives --set terraform /usr/share/terraform/1/bin/terraform
+
+ARG ATMOS_VERSION
+RUN apt-get update && apt-get install -y --allow-downgrades \
+  atmos="${ATMOS_VERSION}-*" \
+  rainbow-text \
+  kubectl
```
Comment (Nuru) [inline examples/complete/Dockerfile:34]: Do not include `rainbow-text` or `kubectl` in default/example Dockerfile. Users do not need `rainbow-text` and should install a version of `kubectl` that is compatible with their version of Kubernetes.

File: internal/exec/helmfile.go [hunk 4]
```diff
@@ -33,22 +33,30 @@ func ExecuteHelmfile(cmd *cobra.Command, args []string) error {
...
-			info.Component,
+			finalComponent,
 			path.Join(c.ProcessedConfig.HelmfileDirAbsolutePath, info.ComponentFolderPrefix),
 		))
 	}
 
+	// Check if the component is allowed to be provisioned (`metadata.type` attribute)
+	if (info.SubCommand == "sync" || info.SubCommand == "apply" || info.SubCommand == "deploy") && info.ComponentIsAbstract {
+		return errors.New(fmt.Sprintf("Abstract component '%s' cannot be provisioned since it's explicitly prohibited from being deployed "+
+			"with 'metadata.type: abstract' attribute", path.Join(info.ComponentFolderPrefix, info.Component)))
```
Comment (Nuru) [inline internal/exec/helmfile.go:57]:
```suggestion
			"by 'metadata.type: abstract' attribute", path.Join(info.ComponentFolderPrefix, info.Component)))
```

File: pkg/config/utils.go (hunk 1 above)
Comment (Nuru) [inline pkg/config/utils.go:253]: I will leave it up to you, but I suggest that if you are going to keep this around, you make it a bit more generic, and have it take arguments. Maybe call it `makeRelativePath` and take `basePath` and `targetPath` as arguments?

File: pkg/config/utils.go [hunk 5]
```diff
@@ -247,6 +248,83 @@ func checkConfig() error {
...
+		Config.Components.Terraform.BasePath = configAndStacksInfo.TerraformDir
+		color.Cyan(fmt.Sprintf("Using command line argument '%s' as terraform directory", configAndStacksInfo.TerraformDir))
+	}
+	if len(configAndStacksInfo.HelmfileDir) > 0 {
+		Config.Components.Helmfile.BasePath = configAndStacksInfo.HelmfileDir
+		color.Cyan(fmt.Sprintf("Using command line argument '%s' as helmfile directory", configAndStacksInfo.HelmfileDir))
+	}
+	if len(configAndStacksInfo.ConfigDir) > 0 {
+		Config.Stacks.BasePath = configAndStacksInfo.ConfigDir
+		color.Cyan(fmt.Sprintf("Using command line argument '%s' as stacks directory", configAndStacksInfo.ConfigDir))
```
Comment (Nuru) [inline pkg/config/utils.go:303]:
```suggestion
		color.Cyan(fmt.Sprintf("Using command line argument '%s' as config directory", configAndStacksInfo.ConfigDir))
```

**Review by Nuru — CHANGES_REQUESTED** [review]

File: pkg/config/utils.go (hunk 5 above)
Comment (aknysh) [reply pkg/config/utils.go:303]: this is  still stacks dir, the old atmos called it config, it's here for backwards compatibility

**Pull request merged**